src/sql/workbench/services/query/common/queryManagement.ts

```diff
@@ -266,6 +268,15 @@ export class QueryManagementService implements IQueryManagementService {
 		});
 	}
 
+	public changeConnectionUri(newUri: string, oldUri: string): Promise<void> {
+		let item = this._queryRunners.get(oldUri);
+		this._queryRunners.set(newUri, item);
```

Comment:
And if we have the new URI already that should also probably be an error like in queryModel